File: tests/prime_numbers.py
import math

def mod(a, b):
    if a < b:
        return a
    else:
        return a - b*(a//b)

def isPrime(x):
    for i in range(3, math.floor(x/2)):
            if mod(x, i) == 0:
                return False
    return True


def getNthPrimeNumber(n):
    count = 4
    prime = 7
    while count < n:
        prime += 2
        if (isPrime(prime)):
            count += 1
    return prime

p = getNthPrimeNumber(1000000)
#print(p)

File: tests/prime_numbers.js
function mod(a, b) {
    if (a < b) {
        retuen a;
    } else {
        return a - b * Math.floor(a / b);
    }
}

function isPrime(x) {
    for (var i=3; i<Math.floor(x/2); i++) {
        if ((x % i) == 0) {
            return false;
        }
    }
    return true;
}

function getNthPrimeNumber(n) {
    var count = 1;
    var prime = 2;
    while (count < n) {
        prime += 2;
        if (isPrime(prime)) {
            count++;
        }
    }
    return prime;
}

var p = getNthPrimeNumber(1000000);
//console.log(p)
